cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -284,9 +284,9 @@ ReconciliationState createReconciliationState(Reconciliation reconciliation, Kaf
     class ReconciliationState {
         private final String namespace;
         private final String name;
-        private final Kafka kafkaAssembly;
-        private final Reconciliation reconciliation;
-        private final KafkaMetadataStateManager kafkaMetadataStateManager;
+        /* test */ final Kafka kafkaAssembly;
+        /* test */ final Reconciliation reconciliation;
+        /* test */ final KafkaMetadataStateManager kafkaMetadataStateManager;
```

Comment:
Why is this needed?

Reply:
Because of the mocking of the reconciliation state when it needs to pass them to the mocked kafka reconciler.

Reply:
doesn't that beat the purpose of the test? The purpose of the test is to see how it flows -> but if you manipulate the internal states in between it does not fulfill that purpose I guess?

Reply:
As discussed offline, I will put together a PR on main to provide Kafka agent client via the ResourceSupplier and then using it here.

Reply:
I opened this one https://github.com/strimzi/strimzi-kafka-operator/pull/9723
When that one is merged, I will simplify the mocking here.